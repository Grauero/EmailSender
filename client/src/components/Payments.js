import React from 'react';
import PropTypes from 'prop-types';
import StripeCheckout from 'react-stripe-checkout';
import { connect } from 'react-redux';

import { handleToken } from '../store/actions';

const Payments = ({ handleToken }) => (
  <StripeCheckout
    name="EmailSender"
    description="$5 for 5 email credits"
    amount={500}
    token={token => handleToken(token)}
    stripeKey={process.env.REACT_APP_STRIPE_KEY}
  >
    <button type="button" className="btn">
      Add Credits
    </button>
  </StripeCheckout>
);

Payments.propTypes = {
  handleToken: PropTypes.func.isRequired
};

export default connect(
  null,
  { handleToken }
)(Payments);
