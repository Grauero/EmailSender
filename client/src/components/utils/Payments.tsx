import React from 'react';
import { Dispatch, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import StripeCheckout from 'react-stripe-checkout';

import { handleToken } from '../../store/actions';

export interface IPayments extends ReturnType<typeof mapDispatchToProps> {}

const Payments: React.FC<IPayments> = props => (
  <StripeCheckout
    name="EmailSender"
    description="$5 for 5 email credits"
    amount={500}
    token={token => props.handleToken(token)}
    stripeKey={process.env.REACT_APP_STRIPE_KEY || ''}
  >
    <button type="button" className="btn">
      Add Credits
    </button>
  </StripeCheckout>
);

const mapDispatchToProps = (dispatch: Dispatch) => {
  return bindActionCreators({ handleToken }, dispatch);
};

export { Payments };
export default connect(
  null,
  mapDispatchToProps
)(Payments);
