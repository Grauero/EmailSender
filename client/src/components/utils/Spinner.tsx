import React, { CSSProperties } from 'react';

import spinner from './spinner.gif';

const Spinner = () => {
  const spinnerStyle: CSSProperties = {
    width: '100px',
    margin: 'auto',
    display: 'block'
  };

  return (
    <div>
      <img src={spinner} alt="Loading..." style={spinnerStyle} />
    </div>
  );
};

export default Spinner;
