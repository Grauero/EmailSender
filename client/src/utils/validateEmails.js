const emailPattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

function validateEmails(emails = '') {
  const invalidEmails = emails
    .split(',')
    .map(email => email.trim())
    .filter(email => email !== '' && !emailPattern.test(email)); 

  if (invalidEmails.length) {
    return `These emails are invalid ${invalidEmails}`;
  }

  return null;
}

export default validateEmails;
