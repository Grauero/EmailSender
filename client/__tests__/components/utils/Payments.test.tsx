import 'jest';
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Payments } from '../../../src/components/utils/Payments';

const component = shallow(<Payments handleToken={jest.fn()} />);

test('matches snapshot', () => expect(toJSON(component)).toMatchSnapshot());
