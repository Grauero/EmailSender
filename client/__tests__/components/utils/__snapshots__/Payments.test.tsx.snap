// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ReactStripeCheckout
  ComponentClass="span"
  amount={500}
  className="StripeCheckout"
  description="$5 for 5 email credits"
  label="Pay With Card"
  locale="auto"
  name="EmailSender"
  reconfigureOnUpdate={false}
  stripeKey=""
  token={[Function]}
  triggerEvent="onClick"
>
  <button
    className="btn"
    type="button"
  >
    Add Credits
  </button>
</ReactStripeCheckout>
`;
